### PR TITLE
Cleanly teardown websocket

### PR DESCRIPTION
**Pull request by enykeev**
* **Please check if the PR fulfills these requirements**
- [x] Tests have been conducted on the PR
- [ ] Docs have been added / updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Bug fix

* **What is the current behavior?** (You can also link to an open issue here)

On KeyboardInterrupt, bot closes websocket connection without awaiting the results resulting in warning:
```
RuntimeWarning: coroutine 'WebSocketCommonProtocol.close' was never awaited
  self._websocket.close()
```

* **What is the new behavior (if this is a feature change)?**

No warning messages produced on KeyboardInterrupt

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

no

### CODE REVIEW

**Comment by A-Trash-Coder** [comment]
This would be great. The message gets annoying.

**Comment by jianyuan** [comment]
It doesn't work if we launch the task using `await bot.start()`. If used this way, we need to call `await self._websocket.close()`.

```python
def main():
	bot = Bot(...)
	await bot.start()

asyncio.run(main())
```

Raises:
```python
raise RuntimeError('This event loop is already running')
```

File: twitchio/websocket.py [hunk 1]
```diff
@@ -612,10 +616,12 @@ async def event_error(self, error: Exception, data: str=None):
         traceback.print_exception(type(error), error, error.__traceback__, file=sys.stderr)
 
     def teardown(self):
+        self._tearingdown = True
+
         if self._bot._webhook_server:
             self._bot._webhook_server.stop()
 
-        self._websocket.close()
+        self.loop.run_until_complete(self.loop.create_task(self._websocket.close()))
```
Comment (IAmTomahawkx) [inline twitchio/websocket.py:624]: This won't work if the loop is running.

Comment (IAmTomahawkx) [inline twitchio/websocket.py:624]: This won't work if the loop is still running

**Review by IAmTomahawkx — CHANGES_REQUESTED** [review]

**Comment by enykeev** [comment]
I will not be returning to this PR after a year. Feel free to take over or close.

**Comment by EvieePy** [comment]
Sorry we haven't looked at this it's been a very busy time for me. This has actually been fixed in 2.0 aswell. Thanks for the PR.

**Pull request closed without merging**